Maple2.Server.Game/Manager/Config/BuffManager.cs

```diff
@@ -0,0 +1,148 @@
+using Maple2.Server.Game.Packets;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager.Config;
+
+public class BuffManager {
+    #region ObjectId
+    private int idCounter;
+
+    /// <summary>
+    /// Generates an ObjectId unique to this specific actor instance.
+    /// </summary>
+    /// <returns>Returns a local ObjectId</returns>
+    private int NextLocalId() => Interlocked.Increment(ref idCounter);
+    #endregion
+    private readonly IActor actor;
+    public ConcurrentDictionary<int, Buff> Buffs { get; } = new();
+    private readonly ILogger logger = Log.ForContext<BuffManager>();
+
+    public BuffManager(IActor actor) {
+        this.actor = actor;
+    }
+
+    public void Initialize() {
+        // Load buffs that are not broadcasted to the field
+        if (actor is FieldPlayer player) {
+            player.Session.Config.Skill.UpdatePassiveBuffs(false);
+            foreach (Item item in player.Session.Item.Equips.Gear.Values) {
+                foreach (ItemMetadataAdditionalEffect buff in item.Metadata.AdditionalEffects) {
+                    AddBuff(actor, actor, buff.Id, buff.Level, false);
+                }
+
+                // Check for buffs on gemstones
+                if (item.Socket?.Sockets is not null && item.Socket.Sockets.Length > 0) {
+                    foreach (ItemGemstone? gemstone in item.Socket?.Sockets!) {
+                        if (gemstone == null || !player.Session.ItemMetadata.TryGet(gemstone.ItemId, out ItemMetadata? metadata)) {
+                            continue;
+                        }
+
+                        foreach (ItemMetadataAdditionalEffect buff in metadata.AdditionalEffects) {
+                            AddBuff(actor, actor, buff.Id, buff.Level, false);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    public void LoadFieldBuffs() {
+        // Lapenshards
+        // Game Events
+        AddMapBuffs();
+    }
+
+    public void AddBuff(IActor caster, IActor owner, int id, short level, bool notifyField = true) {
+        if (Buffs.TryGetValue(id, out Buff? existing)) {
+            if (level > existing.Level) {
+
+            }
+            if (!existing.Stack()) {
+                return;
+            }
+            if (notifyField) {
+                owner.Field.Broadcast(BuffPacket.Update(existing));
+            }
+            return;
+        }
+
+        if (!owner.Field.SkillMetadata.TryGetEffect(id, level, out AdditionalEffectMetadata? additionalEffect)) {
+            logger.Error("Invalid buff: {SkillId},{Level}", id, level);
+            return;
+        }
+
+        // if (!SkillUtils.CheckCondition(additionalEffect.Condition, caster, owner, this)) {
+        //     Console.WriteLine($"Ignore buff {id}");
+        //     return;
+        // }
+
+        var buff = new Buff(owner.Field, additionalEffect, NextLocalId(), caster, owner);
+        if (!Buffs.TryAdd(id, buff)) {
```

Comment:
Another note for the future: if `casterIndividualEffect="1"` then you should track who casted the effect & only check for buffs placed by the same caster. For example effect `10200112` - `Deep Cuts` can be applied by multiple berserkers simultaneously.